From 0f6318c9eb77cbd3de89a3dd16dcc4a76ee9ef71 Mon Sep 17 00:00:00 2001
From: Dynamic Devices <engineering@dynamicdevices.co.uk>
Date: Fri, 11 Sep 2026 13:40:00 +0000
Subject: [PATCH] lxc: drop AppArmor key when AppArmor is unavailable

Foundries LmP builds LXC without AppArmor support.  Leaving
lxc.apparmor.profile in the generated configuration makes LXC reject the
whole file, even when its value is "unconfined".  Remove the setting when
Waydroid's existing AppArmor detection reports that the feature is absent.

Upstream-Status: Pending
Signed-off-by: Dynamic Devices <engineering@dynamicdevices.co.uk>
---
 tools/helpers/lxc.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tools/helpers/lxc.py b/tools/helpers/lxc.py
index d483003..846423d 100644
--- a/tools/helpers/lxc.py
+++ b/tools/helpers/lxc.py
@@ -164,6 +164,9 @@ def set_lxc_config(args):
     if get_apparmor_status(args):
         command = ["sed", "-i", "-E", "/lxc.aa_profile|lxc.apparmor.profile/ s/unconfined/{}/g".format(LXC_APPARMOR_PROFILE), lxc_path + "/config"]
         tools.helpers.run.user(args, command)
+    else:
+        command = ["sed", "-i", "-E", "/lxc.aa_profile|lxc.apparmor.profile/d", lxc_path + "/config"]
+        tools.helpers.run.user(args, command)
 
     nodes = generate_nodes_lxc_config(args)
     config_nodes_tmp_path = args.work + "/config_nodes"
-- 
2.51.0
